### DISCUSSION TITLE
@value works equivalently to value keyword

### DISCUSSION THREAD

**Discussion by osamakawish** [dotnet/csharplang#5372] [Q&A] [answered]
Not sure if this is a bug or issue to be reported or fixed. But it's a little annoying when I try to set a variable name to `@value`, and the property in the setter does nothing different and effectively treats it as if it's `value`.

Basically this:

```c#
private int @value;
public int Value { get => @value; set => value = @value; }
```

causes issues. Code is a bit more complicated than above though, but it's the simplest example I could come up with.

**Comment by svick** [comment, marked as answer]
[Per the spec, `value` is not technically a keyword:](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes?WT.mc_id=DOP-MVP-5002735#accessors)

> The implicit parameter of a `set` accessor is always named `value`.

This means your setter basically behaves as if it was a method with the signature `public void SetValue(int value)` and you can't use `@value` to refer to the field in such a method either. [You can use `this.value`, though:](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQ6BuBDMACAYzwF48oBTAdzwGEAKASgG510CA6ANWwBsJySeACwIWaOEgCcdDtz7lmrcQlroA3ujxa8ABzABLHMAH6owPDnljteOAGY8p83P541eAOblzxAHwXefiY8AGdvEn9gAAt9EPZLV1IE8mCAX3RUoA===)

```c#
private int value;
public int Value { get => value; set => this.value = value; }
```

**Reply by jnm2** [reply]
(The same is also true for the implicit `args` parameter in top-level programs)
